src/config/config_poll.rs

```diff
@@ -0,0 +1,78 @@
+use std::sync::atomic::AtomicPtr;
+use std::sync::Arc;
+use std::time::Duration;
+
+use reqwest::Client;
+use tokio::time;
+
+use crate::blueprint::Blueprint;
+use crate::cli::CLIError;
+use crate::config::Config;
+use crate::http::{SchemaLoader, ServerContext};
+use crate::valid::Valid;
+pub struct ConfigLoader {
+  file_path: String,
+  refresh_interval: u64,
+  state: Arc<AtomicPtr<ServerContext>>,
+}
+
+impl ConfigLoader {
+  pub fn new(file_path: String, refresh_interval: u64, state: Arc<AtomicPtr<ServerContext>>) -> Result<Self, CLIError> {
+    Ok(Self { file_path, refresh_interval, state })
+  }
+
+  pub async fn start_polling(&self) {
```

Comment:
should have exponential backoff in case of failures

Reply:
Servers might be facing temporary downtime or might have got a temporary rate limit, in those case throwing exceptions won't be ideal, it checks everything while starting the server, and I think that should be enough

Reply:
We ought to avoid throwing exceptions; instead, they should be recorded as errors. Moreover, continuous polling of a server that has surpassed its rate limit is counterproductive. Implementing an exponential backoff strategy in case of errors is essential to effectively manage and eventually overcome the rate limit issue.